scripts/setRewardsDuration.js

```diff
@@ -0,0 +1,82 @@
+// ES5 style
+const find = require('lodash/find');
+const readlineSync = require('readline-sync');
+const consola = require('consola');
+
+const Multirewards = require('../build/contracts/MultiRewards.json');
+
+const { getConnex } = require('./utils');
+
+const setRewardsDuration = async ({
+  connex,
+  duration,
+  pool,
+  network,
+  rewardToken,
+}) => {
+  const setRewardsDurationABI = find(Multirewards.abi, { name: 'setRewardsDuration' });
+  const setRewardsMethod = connex.thor.account(pool.address).method(setRewardsDurationABI);
+
+  const clause = setRewardsMethod.asClause(rewardToken, duration);
+
+  if (require.main === module && network === 'mainnet') {
+    const input = readlineSync.question("Confirm you want to execute this on the MAINNET? (y/n) ");
+
+    if (input != 'y') process.exit(1);
+  }
+
+  consola.info(`--------------------- Setting rewards duration for: ${pool.pair} ---------------------`);
+  consola.info(`Duration: ${duration} seconds`);
+  consola.log(' ');
+
+  return new Promise(async (resolve, reject) => {
+    try {
+      const { txid } = await connex.vendor.sign('tx', [clause]).request();
+      consola.info(`Transaction: ${txid}`);
+      await connex.thor.ticker().next()
```

Comment:
Actually just doing a `.next()` is slightly dangerous. Sometimes even at the next block the tx receipt is still not ready. The safer option is to wait for it till you have a receipt.

https://github.com/vexchange/fee-collector/blob/22a8d248714fe9b0d1834bf820772fd6ed580959/scripts/SweepDesired.js#L84